pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -441,17 +438,98 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
+	for _, volume := range requestedVolumes {
+		// not the same as len(pvNamesForRequestedVolumes). PV-s for a new pod might not be cached yet.
+		if volume.PersistentVolumeClaim != nil {
+			requestedDiskCount++
+			if v, ok := pvcToPvMap.Load(getQualifiedName(ns, volume.PersistentVolumeClaim.ClaimName)); ok {
+				entry := v.(*pvcToPVEntry)
+				shouldCheckEntry := true
+				for _, filterOutAccessMode := range filterOutAccessModes {
+					if strings.EqualFold(entry.AccessMode, string(filterOutAccessMode)) {
+						shouldCheckEntry = false
+					}
+				}
+				if shouldCheckEntry {
+					if v, ok := pvToDiskNameMap.Load(entry.VolumeName); ok {
+						diskName := v.(string)
+						diskRequestedByPod[diskName] = struct{}{}
+					}
+				}
+			}
+		}
+	}
+	return
+}
+
 func onPvcUpdate(obj interface{}, newObj interface{}) {
 	// Cast the obj to pvc
 	pvc := newObj.(*v1.PersistentVolumeClaim)
-	klog.V(4).Infof("updating pvcToPv mapping cache. PVC: %s PV: %s, AccessMode: %v", pvc.Name, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
-	cacheValue := cacheEntry{VolumeName: pvc.Spec.VolumeName, AccessMode: fmt.Sprintf("%v", pvc.Spec.AccessModes[0])}
-	pvcToPvMapCache.AddOrUpdate(pvc.Name, &cacheValue)
+	addPvcEntry(pvc)
+}
+
+func addPvcEntry(pvc *v1.PersistentVolumeClaim) {
+	qualifiedName := getQualifiedName(pvc.Namespace, pvc.Name)
+	klog.Infof("Updating pvcToPv map cache.PVC: %s PV: %s, AccessMode: %v", qualifiedName, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
+	cacheValue := pvcToPVEntry{VolumeName: pvc.Spec.VolumeName, AccessMode: fmt.Sprintf("%v", pvc.Spec.AccessModes[0])}
+	pvcToPvMap.Store(qualifiedName, &cacheValue)
 }
 
 func onPvcDelete(obj interface{}) {
 	// Cast the obj to pvc
 	pvc := obj.(*v1.PersistentVolumeClaim)
-	klog.V(4).Infof("deleting from pvcToPv mapping cache. PVC: %s PV: %s", pvc.Name, pvc.Spec.VolumeName)
-	pvcToPvMapCache.Delete(pvc.Name)
+	qualifiedName := getQualifiedName(pvc.Namespace, pvc.Name)
+	klog.Infof("Deleting pvcToPv map cache entry. PVC: %s PV: %s", pvc.Name, pvc.Spec.VolumeName)
+	pvcToPvMap.Delete(qualifiedName)
+}
+
+func onPvUpdate(obj interface{}, newObj interface{}) {
+	// Cast the obj to pvc
+	pv := newObj.(*v1.PersistentVolume)
+	addPvEntry(pv)
+}
+
+func addPvEntry(pv *v1.PersistentVolume) {
+	// TODO: azDiskSchedulerExtender doesn't yet support a flag to set manual driver name, so for now, map all csi volume handles
+	if pv.Spec.CSI != nil {
+		if diskName, err := azureutils.GetDiskName(pv.Spec.CSI.VolumeHandle); err == nil {
+			klog.Infof("Updating pvToVolume map cache.PV: %s, DiskName: %v", pv.Name, diskName)
+			pvToDiskNameMap.Store(pv.Name, strings.ToLower(diskName))
+		}
+	}
+}
+
+func onPvDelete(obj interface{}) {
+	// Cast the obj to pvc
+	pv := obj.(*v1.PersistentVolume)
+	klog.Infof("Deleting pvToVolume map cache.PV: %s", pv.Name)
+	pvToDiskNameMap.Delete(pv.Name)
+}
+
+func getQualifiedName(namespace, name string) string {
+	return fmt.Sprintf("%s/%s", namespace, name)
+}
+
+func recover() error {
```

Comment:
Is this used anywhere?

Reply:
Yes, in unit test, for simpleClient, `forceSync` didn't do the job.

Reply:
We could move it to unittest file then. ForceSync will work only for delta fifo implementation, and I doubt that's what the fake client is using.